### ISSUE TITLE
By default provide a safe and cloud compatible saving method

### ISSUE THREAD

**Issue by gagarine** [keepassxreboot/keepassxc#8920]
KeePassXC provide different saving method and backup mechanism (backup before save). The default method seem to create problems when saving to the cloud and access the same files from multiple devices.

Saving password files to the cloud is a common use case and should be supported by default and with the default settings.

"Directly write to the database" is marked as dangerous but without explaining the risk. Why providing a risky method at the first place in the configuration?

In short, KeePassXC should work out of the box and be safe.

As a general note, I think the user experience would benefit from settings simplification.

**Comment by droidmonkey** [comment]
We have a section in our user guide devoted to this discussion. We cannot "simplify" the settings because there is such a wide variety of use cases out there across all three platforms. You have to set up the program to your needs. By default, we work out of the box with every major cloud and personal storage solution. On Linux that is complicated by several factors that necessitate the use of the "Direct Write" option.

**Comment by gagarine** [comment]
Mmmh ok, so their is a bug with icloud ios and macos. It works to write from MacOS, but then the file link is not the same and ios lose it. I will open a bug report.

**Comment by droidmonkey** [comment]
One of the ios apps doesn't handle icloud files very well. They must be using a direct file pointer instead of accessing the file by name. When a safe save is conducted, the original file is deleted and the new one renamed into its place. This is an atomic operation which means there is almost zero possibility for data loss due to power outage or program error (that is why it is "safe"). However, if you access files by their specific pointer then the "new file" will not be associated with that pointer.
